tests/test_cmd_resetdep.py

```diff
@@ -81,6 +81,33 @@ def test_missing_file_dep(self, depfile):
         assert ("failed t2 (Dependent file 'tests/data/missing' does not "
                 "exist.)\n") == got
 
+    def test_missing_dep_and_target(self, depfile, dependency1):
+
+        task_a = Task("task_a", [""],
+                      file_dep=['tests/data/dependency1'],
+                      targets=['tests/data/dependency2'])
+        task_b = Task("task_b", [""],
+                      file_dep=['tests/data/dependency2'],
+                      targets=['tests/data/dependency3'])
+        task_c = Task("task_c", [""],
+                      file_dep=['tests/data/dependency3'],
+                      targets=['tests/data/dependency4'])
+
+        output = StringIO()
+        tasks = [task_a, task_b, task_c]
+        cmd_reset = CmdFactory(ResetDep, outstream=output, task_list=tasks,
+                               dep_manager=depfile)
+        cmd_reset._execute()
+
+        got = [line.strip().split()[:2] for line in output.getvalue().split('\n') if line]
+        got_status = [g[0] for g in got]
```

Comment:
i dont like much logic in tests. It is preferable to hard-code things and repeat yourself.